core/Symbols.cc

```diff
@@ -952,14 +952,18 @@ string Symbol::show(const GlobalState &gs) const {
 
     if (this->owner == core::Symbols::PackageRegistry()) {
         // Pretty print package name (only happens when `--stripe-packages` is enabled)
-        auto nameStr = this->name.show(gs);
-        constexpr string_view packageNameSuffix = "_Package"sv;
-        if (absl::EndsWith(nameStr, packageNameSuffix)) {
```

Comment:
Sorbet is and never has been in the business of doing name-based decisions based on string comparisons, and I don't see why that should stop being the case for packager names.

As such I've redone the support for packager names in this PR to use a new unique name kind, so that asking whether a name is a packager name is a matter of checking a few bits.

Reply:
Good idea. I didn't feel emboldened to do this sort of thing when I first added packager support to Sorbet.